### PR TITLE
WorkspaceCommand: ability to remove workspaces

### PR DESCRIPTION
**Pull request by vmax**

### CODE REVIEW

File: tools/aa/commands/WorkspaceCommand.java [hunk 1]
```diff
@@ -103,6 +113,10 @@ public boolean run(String[] args) {
     String workspacePath = fileUtils.joinPaths(basePath, "ws", workspaceName);
 
     if (force.get()) {
+      if (remove.get()) {
```
Comment (oferb) [inline tools/aa/commands/WorkspaceCommand.java]: Let's put this in `checkFlags()`

File: tools/aa/commands/WorkspaceCommand.java [hunk 2]
```diff
@@ -121,6 +135,16 @@ public boolean run(String[] args) {
         System.err.println(RED_ERROR + "Workspace does not exist");
         return false;
       }
+
+      if (remove.get()) {
+        Path currentPath = Paths.get("").toAbsolutePath();
+        if (currentPath.startsWith(workspacePath)) {
+          System.err.println(RED_ERROR + "Trying to remove active workspace");
```
Comment (oferb) [inline tools/aa/commands/WorkspaceCommand.java]: Why not allow removing active workspace?
In bash, you can delete the current folder.

Comment (vmax) [reply tools/aa/commands/WorkspaceCommand.java]: @oferb but is this a feature and not a bug? 🙂 
apart from `cd`-ing away, you won't be able to do much being left in this state

File: tools/aa/commands/WorkspaceCommand.java [hunk 3]
```diff
@@ -42,93 +43,162 @@
...
     }
 
     args = Arrays.copyOfRange(args, 0, args.length - 1);
-
     Flags.parse(args, WorkspaceCommand.class.getPackage());
-    String basePath = config.getBasePath();
-    String workspacePath = fileUtils.joinPaths(basePath, "ws", workspaceName);
+
+    if (force.get() && remove.get()) {
+      System.err.println(RED_ERROR + "Conflicting options specified");
```
Comment (oferb) [inline tools/aa/commands/WorkspaceCommand.java]: How about:
`Cannot specify both force and remove flags`

File: tools/aa/commands/WorkspaceCommand.java [hunk 4]
```diff
@@ -42,93 +43,162 @@
...
+
+  @Override
+  public boolean run(String[] args) {
+    // Note: System.out gets executed by the calling aa_tool.sh to run commands such as cd.
+    if (!processArgs(args)) {
+      return false;
+    }
+
+    if (mode == ActionMode.CREATE) {
+      if (!createWorkspace()) {
```
Comment (oferb) [inline tools/aa/commands/WorkspaceCommand.java]: How about we put the positive first:
```
if (createWorkspace()) {
  switchWorkspace();
} else {
  return false;
}
```
(also below)

Comment (vmax) [reply tools/aa/commands/WorkspaceCommand.java]: yay, we saved space on two `!` characters 🙂

Comment (oferb) [reply tools/aa/commands/WorkspaceCommand.java]: :-)

**Pull request merged**